Add notification_parser method to Connection

corylutton opened the pull request:
As mentioned in #143 - here is my take on moving the logic from the decorator into a different method to be used in other ways + in the decorator.

Hopefully this is up to par, I tried to get the documentation/type info in there as the rest of the code had it... not really that familiar with this process (Github) so could always take another run at it if needed.  I left all the tests alone as all the notification decorator tests just call this function, was not sure if I should poke into that at all or what would make sense to change that into since the decorator now calls this function.

stlehmann commented on pyads/ads.py:
```diff
@@ -633,7 +633,9 @@ def ams_netid(self):
     def ams_netid(self, netid):
         # type: (str) -> None
         if self._open:
-            raise AttributeError("Setting netid is not allowed while connection is open.")
```
No need to change that. It is not related to the issue.

stlehmann commented on pyads/ads.py:
```diff
@@ -645,7 +647,9 @@ def ams_port(self):
         # type: (int) -> None
         if self._open:
-            raise AttributeError("Setting port is not allowed while connection is open.")
+            raise AttributeError(
```
Same as above

stlehmann commented on pyads/ads.py:
```diff
@@ -1107,44 +1111,92 @@ def notification_decorator(func):
-                    else:
-                        # invalid size
-                        value = None
+    def notification_parser(
```
I would prefer the function name to be a verb because it is no object but does something. Maybe something like "parse_notification"

corylutton replied:
Makes sense, parse_notification sounds good to me

corylutton commented:
Addressed your comments above, I think I got it now without additional unrelated changes.  Not sure if there is a way to combine those three commits into one or if it matters.  If I can do anything else to make it better please let me know... long time programmer but only 3rd time doing a github pull request.

stlehmann commented on pyads/ads.py:
```diff
@@ -1107,44 +1107,92 @@ def notification_decorator(func):
+        >>> # Create callback function that prints the value
+        >>> def mycallback(notification, data):
+        >>>     data_type = tag[data]
+        >>>     handle, timestamp, value = plc.notification_parser(notification, data_type)
```
I suppose this should be plc.parse_notification
```suggestion
        >>>     handle, timestamp, value = plc.parse_notification(notification, data_type)
```

corylutton replied:
Documentation adjusted.

stlehmann commented:
Great work. Thanks for this 🎉
